test/extended/testdata/build-extended/bc-scripts-in-the-image.json

```diff
@@ -0,0 +1,34 @@
+{
+  "kind": "BuildConfig",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "java-extended-build-from-image"
+  },
+  "spec": {
+    "source": {
+      "git": {
+        "uri": "https://github.com/php-coder/java-maven-hello-world.git",
+        "ref": "s2-assemble-only"
+      }
+    },
+    "strategy": {
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "wildfly:10.0",
+          "namespace": "openshift"
+        },
+        "runtimeImage": {
+          "kind": "DockerImage",
+          "name": "phpcoder/hello-world-runtime:0.2"
```

Comment:
and also it's not clear from these tests that you're both testing the annotation *and* the script location.  i would expect an explicit test that tests the annotation/mapping config field.  Or at least some sort of indication in the test that it is testing that.  (beyond just the presence/absence of the runtimeartifacts field in the config)